js/store.js

```diff
@@ -345,6 +350,10 @@ const store = {
     return true;
   },
 
+  isAlsoAFirstParty(thirdParty) {
```

Comment:
Should this function work on both fp+tp's instead of just being given a node. This could check both keys.

Reply:
If I am understanding you correctly... The default behavior is to overwrite the value of `firstPaty` (whether `true` or `false`) with the new request data, so we only need to worry about the case when a first party later becomes a third party (we want to preserve `firstParty` as `true`), and that's what this method checks.

I am open to a less confusing way of doing this!

Reply:
I think we need to cope when a either type become a dual. So I think this check should happen for both on update.